### PR TITLE
move onResolversCalled to the top level

### PR DESCRIPTION
**Pull request by n1ru4l**

### CODE REVIEW

File: .changeset/fresh-toes-hang.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+-    return {
+-      onResolversCalled() {}
+-    }
+  }
+}
+```
+
+We highly recommend avoiding to use any plugins that use `onResolversCalled` within your production environment as it has severe impact on the performance of the individual resolver functions within your schema.
+
+The schema resolver functions are now ONLY wrapped if any plugin in your envelop setup uses the `onResolversCalled``hook.
```
Comment (dotansimha) [inline .changeset/fresh-toes-hang.md]: duplicate `

**Comment by kamilkisiela** [comment]
@n1ru4l could you rebase? I added benchmarks to main branch

**Review by dotansimha — APPROVED** [review]
LGTM. Thanks @n1ru4l ! 

~rebase to include @kamilkisiela 's changes for benchmarks?~

**Comment by n1ru4l** [comment]
closed in favor of https://github.com/dotansimha/envelop/pull/1236

**Pull request closed without merging**